src/Task.cc

```diff
@@ -841,15 +841,20 @@ TrapReasons Task::compute_trap_reasons() {
 const Task::ThreadLocals& Task::fetch_preload_thread_locals() {
   if (tuid() == as->thread_locals_tuid()) {
     if (as->has_mapping(AddressSpace::preload_thread_locals_start())) {
-      memcpy(thread_locals,
-             as->mapping_of(AddressSpace::preload_thread_locals_start())
-                 .local_addr,
-             PRELOAD_THREAD_LOCALS_SIZE);
-    } else {
-      // The mapping might have been removed by crazy application code.
-      // That's OK, assuming the preload library was removed too.
-      memset(&thread_locals, 0, sizeof(thread_locals));
+      uint8_t* local_addr =
+          as->mapping_of(AddressSpace::preload_thread_locals_start())
+              .local_addr;
+      if (local_addr) {
```

Comment:
This is a bit scary. It's possible that the mapping was replaced by something else that gets a local mapping, and then this code will run and we'll scribble over it. That would be pretty hard to debug. Can we additionally check that the name starts with `preload_thread_locals`?